Future-Hope/Future-Hope/OnBoardingViews/EmailSignUpViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  EmailSignUpViewController.swift
+//  Future-Hope
+//
+//  Created by Hector Steven on 8/20/19.
+//  Copyright © 2019 Hector Steven. All rights reserved.
+//
+
+import UIKit
+
+import Firebase
+
+import MaterialComponents.MaterialTextFields
+import MaterialComponents.MaterialButtons
+
+class EmailSignUpViewController: UIViewController {
+
+	
+	@IBOutlet var fullNameTextField: MDCTextField!
+	@IBOutlet var emailTextFields: MDCTextField!
+	@IBOutlet var passwordTextField: MDCTextField!
+	@IBOutlet var confirmPasswordTextField: MDCTextField!
+	
+
+	override func viewDidLoad() {
+        super.viewDidLoad()
+		
+//		navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Test", style: .plain, target: self, action: #selector(Test))
+		
+    }
+	
+//	@objc func Test() {
+//		fullNameTextField.text = "Hector Vill"
+//		emailTextFields.text = "hectorsvill@gmail.com"
+//		passwordTextField.text = "Pass1234"
+//		confirmPasswordTextField.text = "Pass1234"
+//
+//	}
+	
+	@IBAction func submitButtonPressed(_ sender: MDCButton) {
+		
+		
+		guard let fullName = fullNameTextField.text,
+			let email = emailTextFields.text,
+			let password = passwordTextField.text,
+			let confirmPassword = confirmPasswordTextField.text else { return }
+		
+		if !checkTextIsSafe(fullName: fullName, password: password, confirmPassword: confirmPassword) {
+			return
+		}
+
+		Auth.auth().createUser(withEmail: email, password: password) { authResult, error in
+			if let error = error {
+				
+				NSLog("Error with Auth Sign In email/password: \(error)\n With authResult: \(authResult.debugDescription)")
+				let alertController = ApplicationController().simpleActionSheetAllert(with: "Error Loging In with email/password with Sign Up", message: "Please Try Again!")
+				self.present(alertController, animated: true)
+				
+				return
+			}
+			
+			print("\(fullName) - \(email) - \(password)")
+			// MARK: Save to User Defaults
+			// MARK: send user data to firestore and new view
+			// MARK: segue to app
+			
+		}
+	}
+	
+	@IBAction func cancelButtonPressed(_ sender: MDCButton) {
+		navigationController?.popViewController(animated: true)
+	}
+	
+}
+
+extension EmailSignUpViewController{
+	
+	private func checkTextIsSafe(fullName: String, password: String, confirmPassword: String) -> Bool{
```

Comment:
Really nice helper method. 🤜🤛 I like that you put it inside of an extension.

Reply:
Yeah. It's easier to read and a little neater!